src/main/java/duke/ui/Parser.java

```diff
@@ -0,0 +1,116 @@
+     * @return The Command that user wants to execute.
+     * @throws DukeException If there is an invalid command.
+     * @throws DateTimeParseException If there is an error in Date and Time input.
+     */
+    public static Command parse(String command) throws DukeException, DateTimeParseException {
+        switch (getInput(command)) {
+        case BYE:
+            return new ByeCommand();
+            // Fallthrough
+        case LIST:
+            return new ListCommand();
+            // Fallthrough
+        case MARK:
+            int markNum = Integer.parseInt(command.replace("mark ", ""));
+            return new MarkCommand(markNum - 1);
+            // Fallthrough
+        case UNMARK:
+            int unMarkNum = Integer.parseInt(command.replace("unmark ", ""));
+            return new UnMarkCommand(unMarkNum - 1);
+            // Fallthrough
+        case TODO:
+            String tDes = command.replace("todo", "");
+            return new AddCommand(new Todo(tDes));
+            // Fallthrough
+        case EVENT:
+            String[] eDes = command.replace("event", "").split(" /at ");
+            return new AddCommand(new Event(eDes[0], LocalDate.parse(eDes[1])));
+            // Fallthrough
+        case DEADLINE:
+            String[] dDes = command.replace("deadline", "").split(" /by ");
+            return new AddCommand(new Deadline(dDes[0], LocalDate.parse(dDes[1])));
+            // Fallthrough
+        case DELETE:
+            int delNum = Integer.parseInt(command.replace("delete ", ""));
+            return new DeleteCommand(delNum - 1);
+            // Fallthrough
+        case FIND:
+            String[] findKeys = command.replace("find ", "").split(" ");
+            return new FindCommand(findKeys);
+            // Fallthrough
+        default:
+            throw new DukeException("I'm sorry, but I don't know what that means :-(");
+            // Fallthrough
+        }
+    }
+
+    /**
+     * Returns enum Inputs which program can understand.
+     *
+     * @param command The user input.
+     * @return An enum that the program understands.
+     */
+    private static Inputs getInput(String command) {
+        if (command.equals("bye")) {
+            return Inputs.BYE;
+        } else if (command.equals("list")) {
+            return Inputs.LIST;
+        } else if (command.startsWith("mark")) {
+            return Inputs.MARK;
+        } else if (command.startsWith("unmark")) {
+            return Inputs.UNMARK;
+        } else if (command.startsWith("todo")) {
+            return Inputs.TODO;
+        } else if (command.startsWith("deadline")) {
+            return Inputs.DEADLINE;
+        } else if (command.startsWith("event")) {
+            return Inputs.EVENT;
+        } else if (command.startsWith("delete")) {
+            return Inputs.DELETE;
+        } else if (command.startsWith("find")) {
+            return Inputs.FIND;
+        } else {
+            return Inputs.ELSE;
+        }
```

Comment:
Nothing really wrong here but since there are multiple else blocks, perhaps a switch statement is better?